packages/next-on-pages/src/buildApplication/processVercelOutput.ts

```diff
@@ -23,6 +27,8 @@ export async function getVercelStaticAssets(): Promise<string[]> {
 		return [];
 	}
 
+	await deleteNextTelemetryFiles(dir);
+
 	return (
```

Comment:
Why is `getVercelStaticAssets()` necessary? Can't I just deal with the entire directory rather than treeing all the files? This would make copying to the custom outdir much easier since I could just `cp -r .vercel/output/static <outdir>` rather doing it individually for each file.

Reply:
The `getVercelStaticAssets` function is primarily used to collect a list of static assets that can be injected into the worker so that it knows what static assets are available during routing.

Reply:
Ah, okay! Yeah, that's totally fine. I saw it was used before this PR, but I was just wondering what for.

Out of curiosity, if we offered something like `env.ASSETS.exists(request.url)` and promised a fast response telling you if a file existed or not, would that be of interest? Would that simplify things for next-on-pages? Could you get rid of your own static asset manifest?

Reply:
Yeah, an option like that would mean we don't need to inject all the static assets (which also saves a little bit of bundle size).

The crucial part is of course the response time/reliability of it though, because we check for the existence at multiple points, so there's the potential for it to get hit a lot of times depending on the size/complexity of a project.